how-to/features/baremetal.rst

```diff
@@ -0,0 +1,378 @@
+running on it for controlling and deploying baremetal nodes. Those images
+can be built with the `Ironic Python Agent Builder`_, or community-built
+images can be used instead:
+
+::
+
+   # load OpenStack admin credentials.
+   . ~/.openrc
+   unset OS_SYSTEM_SCOPE
+
+   wget https://tarballs.openstack.org/ironic-python-agent/tinyipa/files/tinyipa-master.vmlinuz
+   wget https://tarballs.openstack.org/ironic-python-agent/tinyipa/files/tinyipa-master.gz
+
+   DEPLOY_VMLINUZ_UUID="$(openstack image create tinyipa-deploy-ipmi.vmlinuz --public --disk-format=raw --container-format=bare --file ./tinyipa-master.vmlinuz -f value -c id)"
+   DEPLOY_INITRD_UUID="$(openstack image create tinyipa-deploy-ipmi.initramfs --public --disk-format=raw --container-format=bare --file ./tinyipa-master.gz -f value -c id)"
+
+   openstack image import tinyipa-deploy-ipmi.vmlinuz --method copy-image --store swift
+   openstack image import tinyipa-deploy-ipmi.initramfs --method copy-image --store swift
+
+As an example, let's consider registering a bare metal node into Ironic using
+the IPMI driver:
+
+::
+
+   # considering that the admin has a system-scoped role, as mentioned above.
+   unset OS_PROJECT_NAME OS_PROJECT_DOMAIN_NAME
+   export OS_SYSTEM_SCOPE=all
+
+   # provide values for the following variables:
+   IPMI_ADDR=""
+   IPMI_PORT=""
+   IPMI_USER=""
+   IPMI_PASS=""
+   IRONIC_NETWORK=""
+   NODE_MAC_ADDR=""
+   SWITCH_INFO=""
+   SWITCH_ID=""
+
+   # register the node.
+   chassis_id=$(openstack baremetal chassis create -f value -c uuid)
+   machine_uuid="$(openstack baremetal node create --name ironic-machine1 --driver ipmi --chassis $chassis_id -c uuid -f value)"
+   openstack baremetal node set ironic-machine1 \
+     --resource-class baremetal \
+     --driver-info ipmi_address=$IPMI_ADDR --driver-info ipmi_port=$IPMI_PORT \
+     --driver-info ipmi_username=$IPMI_USER --driver-info ipmi_password=$IPMI_PASS \
+     --driver-info deploy_kernel=$DEPLOY_VMLINUZ_UUID \
+     --driver-info deploy_ramdisk=$DEPLOY_INITRD_UUID \
+     --driver-info cleaning_network=$IRONIC_NETWORK \
+     --driver-info provisioning_network=$IRONIC_NETWORK
+
+   # register a port for the node.
+   port_uuid="$(openstack baremetal port create $NODE_MAC_ADDR --node $machine_uuid -c uuid -f value)"
+   openstack baremetal port set $port_uuid --local-link-connection switch_info=$SWITCH_INFO \
+     --local-link-connection switch_id=$SWITCH_ID --local-link-connection port_id=$NODE_MAC_ADDR
+
+   # validate and provide the node; it should become available.
+   openstack baremetal node validate ironic-machine1
+   openstack baremetal node manage ironic-machine1
+   openstack baremetal node show ironic-machine1
+   openstack baremetal node provide ironic-machine1
+
+.. important::
+   The `IRONIC_NETWORK` network mentioned above must be a network that can reach
+   its associated `ironic-conductor` HTTP and TFTP services, exposed through
+   its Internal Load Balancer IP.
```

Comment:
This is not very clear.

Do you mean to say that the IRONIC_NETWORK must be able to reach the `internal` network? Where Ironic Conductor will have HTTP/TFTP services exposed?

Reply:
Yes. I've updated this section a bit, hopefully it's a bit clearer.